### ISSUE TITLE
Office.AutoShowTaskpaneWithDocument only seems to work when hooked up to a button, and in desktop excel

### ISSUE THREAD

**Issue by EdwinStraver** [OfficeDev/office-js#5235]
I am trying to make sure my add-in which has a custom ribbon, and many commands, will automatically open a default taskpane when the add-in is installed.
I used the recommended  code with Office.AutoShowTaskpaneWithDocument

However, I note it only works when i put it in the action controller of a button. It does NOT work when I attach it to the action controller of a menu item.

In addition, I don't seem to be able to make it work in Excel for the Web.

Are both these issues by design...or.. ?

Edwin

**Comment by guoms1** [comment]
Hi @EdwinStraver<br>
Thank you for sharing the details! To help me investigate this issue more effectively, could you please provide:

- A complete add-in project that reproduces the problem (this would be ideal), or
- A code snippet that demonstrates the issue clearly.

Having this will allow me to quickly reproduce and analyze the behavior you’re observing. I appreciate your time and effort in helping us address this!

Looking forward to your response.<br>
Best regards

**Comment by EdwinStraver** [comment]
No, I dont have a complete add-in project that reproduces the problem....I assume that was a joke.

I follow the documentation from https://learn.microsoft.com/en-us/office/dev/add-ins/develop/automatically-open-on-installation

If I add this code in a manifest file:

<Action xsi:type="ShowTaskpane">
    <TaskpaneId>Office.AutoShowTaskpaneWithDocument</TaskpaneId>
    <SourceLocation resid="Contoso.Taskpane.Url" />
</Action>

it works IF the code is associated with a BUTTON. It does NOT work when the code is hooked up to a MENU.
It also does NOT seem to work when I add the add-in to Excel for the web; no taskpane opens automatically.

**Comment by guoms1** [comment]
Hi @EdwinStraver,

Thank you for reaching out and sharing the details of the issues you're experiencing with the AutoShowTaskpaneWithDocument feature in your Office JavaScript Add-in.

Regarding the problems you've described, I have the following observations:

1. Excel for the Web: In my testing, the task pane automatically opens as expected when using the provided Excel manifest. This suggests that the feature should work correctly in Excel Online when the manifest is properly configured.
2. Menu Items: Based on your description, it appears that the task pane opens correctly when triggered by a button but not when associated with a menu item. In my tests, both buttons and menu items behave similarly, successfully opening the task pane.

I was unable to replicate the issues on my end, which might be due to not following the exact steps you used. To assist you further, I have attached two manifest files ([auto-open-task-pane-manifests.zip](https://github.com/user-attachments/files/18394244/auto-open-task-pane-manifests.zip))—one for Word and one for Excel. Each manifest includes configurations for both buttons and menu items. You can side-load these manifests directly to test and verify the functionality of the AutoShowTaskpaneWithDocument feature in both applications.

If you continue to encounter issues after testing with these manifests, it would be incredibly helpful if you could share a minimal sample project that reproduces the problem. Having a simplified version of your add-in will allow us to understand the issue more accurately and work towards a solution more efficiently.

Please feel free to reach out if you have any questions or need further assistance with the manifest files. I'm here to help and ensure your add-in functions seamlessly across all platforms.

Looking forward to your feedback!

Best regards,

**Comment by EdwinStraver** [comment]
Hi gumos1,

Thank you for your thoughtful reply.  However, your manifest files exhibit exactly what I tried to explain:  I centrally deployed your addin (auto-open-task-pane-xl.xml), and sure enough, as soon as I install the add-in the taskpane opens.

but this is because you have the action item hooked up to both a button and a menu. When I remove the button section:


                <Control xsi:type="Button" id="Contoso.TaskpaneButton">
                  <Label resid="Contoso.TaskpaneButton.Label" />
                  <Supertip>
                    
                    <Title resid="Contoso.TaskpaneButton.Label" />
                    
                    <Description resid="Contoso.TaskpaneButton.Tooltip" />
                  </Supertip>
                  <Icon>
                    <bt:Image size="16" resid="Contoso.tpicon_16x16" />
                    <bt:Image size="32" resid="Contoso.tpicon_32x32" />
                    <bt:Image size="80" resid="Contoso.tpicon_80x80" />
                  </Icon>

                  
                  <Action xsi:type="ShowTaskpane">
                    
                    <TaskpaneId>Office.AutoShowTaskpaneWithDocument</TaskpaneId>
                    
                    <SourceLocation resid="Contoso.Taskpane.Url" />
                  </Action>
                </Control>

and then again install the addin, it installs fine, but the taskpane does NOT automatically open in Excel.

I have attached the modified manifest file, so you can confirm this as well.
[auto-open-task-pane-xl.zip](https://github.com/user-attachments/files/18400236/auto-open-task-pane-xl.zip)

Greetings, Edwin

**Comment by guoms1** [comment]
Hi @EdwinStraver 

Thank you for your patience and for working through the issue with me.

I wanted to let you know that using the manifest.xml files I provided, I was able to successfully auto-open the task pane in Excel Online. To help us pinpoint the difference, I’ve recorded a short [video](https://github.com/user-attachments/assets/62e471ae-7ae0-490c-8e20-678cc1b7c76e) demonstrating this behavior on my end.

Could you please take a look at the video and compare it with yours? This might help us identify any discrepancies that could be causing the issue you’re experiencing.

If you still encounter problems, sharing the video along with any additional details would be incredibly helpful. I'm here to assist you further and ensure everything works smoothly.

Looking forward to your feedback!

**Comment by EdwinStraver** [comment]
I added the video.

https://github.com/user-attachments/assets/38d6959a-43f6-44a0-b982-44539291f6f5

**Comment by Dhoot** [comment]
Closing this due to inactivity @EdwinStraver feel free to reopen it.
